parse.go

```diff
@@ -59,13 +60,17 @@ func Parse(r io.Reader, opts ...ParseOption) (*parser.Module, *parser.FileBuffer
 
 	name := lexer.NameOfReader(r)
 	if name == "" {
-		name = "<stdin>"
+		name = "/dev/stdin.hlb"
 	}
 	r = &parser.NewlinedReader{Reader: r}
 
 	fb := parser.NewFileBuffer(name)
 	r = io.TeeReader(r, fb)
 
+	if filepath.Ext(name) != ".hlb" {
+		return parser.ParseDockerfile(r)
```

Comment:
If I'm reading this correctly, it treats every file that doesn't have a `.hlb` extension as a dockerfile. Should we be more explicit and limit it to `name == "Dockerfile" || filepath.Ext(name) == ".dockerfile"`?

Reply:
Unfortunately, Dockerfiles do not have any conventions for naming.

Reply:
Introducing an extension may now require users to rename their Dockerfiles which may break their pipelines built on top.
I'll lean towards maybe a scheme within the HLB import, but I'm still on the fence.

Examples:
```sh
hlb run -t foo dockerfile://TestDockerfile
```

```hlb
import dockerfile from "dockerfile://TestDockerfile"
```

Reply:
It does lose some of the magic like `docker build` that people are used to though.

Reply:
I was thinking about this yesterday and I realized what we need here are _options_ for `import`. For example,

```
import f1 from "./Dockerfile" with dockerfileSyntax
# or
import f2 from "./Dockerfile.foo" with option {
  syntax "dockerfile"
}
```

The first option is short and sweet, the second is explicit and verbose. But they're both parseable with no syntax changes, they don't encode semantics into the string so we can annotate the CST entries, and they're forward-compatible with hlbv2 or whatever other syntax we need in the future, including access to experimental Dockerfile syntax as eg, `with experimentalDockerfileSyntax`.

Reply:
Hey I think that's a neat idea!
You could just do `import f1 from "./Dockerfile" with syntax("dockerfile")` too.

That said, this specific PR is using the `Dockerfile2LLB` as a library, so we won't be able to adjust the syntax other than through the Dockerfile's syntax directive for the gateway (e.g. `# syntax=docker/dockerfile:experimental`). Due to this, we may even need to display the Dockerfile syntax version alongside `hlb version`.

Reply:
The issue with an opaque syntax option is that we don't know if its a Dockerfile (or what version), or some other frontend. And we currently don't allow invoking an imported module as a function. For Dockerfiles, we can import their stages as exported identifiers, but otherwise there aren't any.

1. Do we allow invoking imports? (My opinion: no)
```hlb
import f1 from "./foobar" with syntax("foobar")
fs default() { f1 }
```
2. Do we invoke imports with `gateway.v0`? (Special case for Dockerfile to parse stages?)

Reply:
I think for (1) you're right, it feels "weird" to directly invoke the import, which would otherwise be a module identifier. Because Dockerfile stages don't require names, it might be good to define some automatic naming convention eg, the first stage is synthesized as `f1._1`, and maybe the default stage (ie, last stage) is `f1.default` in addition to whatever user-defined names the stages might have.

As for (2) well I hadn't considered it, but that _would_ be a pretty badass feature. It should probably follow the same naming conventions as the Dockerfile parser, though. 
So given `import f2 from "./build.earthly" with frontend("earth.ly/foo")` maybe its used as `f2.default` for a single Ref, and `f2._1, f2._2, ...` for multiple returned Refs.